crates/zkvm/zkvm/src/lib.rs

```diff
@@ -36,6 +48,68 @@ impl From<&Proof> for Vec<u8> {
     }
 }
 
+impl From<ProofWithMetadata> for Proof {
+    fn from(metadata: ProofWithMetadata) -> Self {
+        metadata.proof
+    }
+}
+
+/// Validity proof generated by the `ZKVMHost` along with some additional information
+#[derive(
+    Debug,
+    Clone,
+    Default,
+    Serialize,
+    Deserialize,
+    BorshSerialize,
+    BorshDeserialize,
+    PartialEq,
+    Eq,
+    Arbitrary,
+)]
+pub struct ProofWithMetadata {
+    /// A unique identifier for the proof. This identifier is consistent for a given fixed input
+    /// and fixed program, meaning that for the same inputs and program, the `id` will always
+    /// remain the same.
+    id: String,
+    proof: Proof,
+    /// An optional identifier that uniquely identifies the proof on a remote server. Unlike `id`,
+    /// this field might vary for the same input and program depending on the context, such as
+    /// server-side processes or external systems. If the proof is generated locally and not
+    /// associated with a remote server, this field will be `None`.
+    remote_id: Option<String>,
```

Comment:
Amazing docstrings, but we need some padding here:
```suggestion
    /// A unique identifier for the proof. This identifier is consistent for a given fixed input
    /// and fixed program, meaning that for the same inputs and program, the `id` will always
    /// remain the same.
    id: String,
    
    proof: Proof,
    
    /// An optional identifier that uniquely identifies the proof on a remote server. Unlike `id`,
    /// this field might vary for the same input and program depending on the context, such as
    /// server-side processes or external systems. If the proof is generated locally and not
    /// associated with a remote server, this field will be `None`.
    remote_id: Option<String>,
```